portality/models/openurl.py

```diff
@@ -124,11 +119,32 @@ def get_result_url(self):
                     if ident is None:
                         ident = journal.id
 
-                # FIXME: the volume and issue are not queried on or validated, so we don't know
-                # for sure at this point whether the user will hit a valid ToC
-                # construct the toc url using the ident, plus volume and issue if they are available
-                jtoc_url = url_for("doaj.toc", identifier=ident, volume=self.volume, issue=self.issue if self.volume else None)
-
+                # If there request has a volume parameter, query for presence of an article with that volume
+                if self.volume:
+                    volume_query = deepcopy(TERMS_SEARCH)
+                    volume_query["size"] = 0
+
+                    issn_term = { "term" : { "index.issn.exact" : ident} }
```

Comment:
A couple of things could go wrong here that we should look at:

1. ident might not be an issn.  For some unknown reason we might have wound up with a journal id, in which case this match may fail when it should have succeeded.

2. Irrespective of (1), journal continuations may have up to 2 issns, and articles may list one or both of them, so a match for one issn won't cover all possible cases.  We need to ask the bibjson for the specific continuation we are interested in for all of its issns (bibjson.issns()), and use these here as part of a "terms" query (not a "term" query).